app/src/main/java/popularmovies/one/fabianreddig/udacity/projectone/movieactivity/MovieListFragment.java

```diff
@@ -51,18 +57,22 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
     public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
         recyclerSetup();
+        if(savedInstanceState==null){
+            refresh(false);
+        }
     }
 
     @Override
-    public void onResume() {
-        super.onResume();
-        refresh(false); // TODO: 5/23/16 Handle resuming to a paginated state
+    public void onDestroy() {
+        super.onDestroy();
+        getMovieListViewModel().detachFromView();
     }
 
     private void loadMovies(int page){
-        addSubscription(apiWrapper.getPopularMovieList(page).compose(RxUtil.singleBackgroundToMainThread())
+        getMovieListViewModel().addSubscription(apiWrapper.getPopularMovieList(page).compose(RxUtil.singleBackgroundToMainThread())
```

Comment:
This will still lead to zombies. A lambda or anonymous class created inside an instance will always have an implicit reference to that instance as long as it modifies that instance's state. A common way to avoid this is to create the anonymous class in a separate object and only reference state / call methods of that object from the anonymous class. In short: dumb views, and smart viewmodels